Misc updates and fixes (#6)
mendersoftware/mender-python-client: README.md

[![Build Status](https://gitlab.com/Northern.tech/Mender/mender-python-client/badges/master/pipeline.svg)](https://gitlab.com/Northern.tech/Mender/mender-python-client/pipelines)

> [![Coverage Status](https://coveralls.io/repos/github/mendersoftware/mender-python-client/badge.svg?branch=master)](https://coveralls.io/github/mendersoftware/mender-python-client?branch=master)

Reviewer: It is up to you. But as the coverage is still very low we could keep it commented, and only un-comment it when this repo is production ready

Author: Fair point actually. I'm uncommenting the coverage for now. Build I'll keep.

Mender Python Client: A Python implementation of the Mender client
==============================================

![Mender logo](mender_logo.png)

## Overview

The _Mender Python Client_ is an API client, which is responsible for
interacting with the Mender server, and downloading the Artifact for a
deployment to a specified location on the device, and then exit until a local
sub-updater on the device has handled the update, reported the update status

> (failed, or success), and re-started the _Python client_ thereafter.

Reviewer: ```suggestion
(failed, or success), and re-started the _Mender Python Client_ thereafter.
```

For consistency with the above.

## Workings

> The _client_ in daemon mode will idle checking for updates at a configureable

Reviewer: "will idle checking" sounds strange to me. Maybe "will idle check" or just "will do periodic checks for updates..."

interval, and download the Artifact for the deployment to a given location on
the device. Then control is passed over to the _sub-updater_ through calling the
script `/usr/share/mender/install <path-to-downloaded-artifact>`.

> It is then the sub-updaters responsibility to unpack the Artifact, and install

Reviewer: Maybe:

```suggestion
It is then the responsibility of the _sub-updater_ unpack the Artifact, and install
```

> it to the passive partition, reboot the device, commit the update (or roll back
> if so is required). Then report the update status through calling

Reviewer: Probably fine, but I would be more explicit about when to roll back:

```suggestion
it to the passive partition, reboot the device, commit the update (or roll back
in case of errors). Then report the update status through calling
```

`mender-python-client report <--success|--failure>`, and then remove the

> lock-file, to have the _Python Client_ start looking for updates again.

Reviewer: ```suggestion
lock-file, to have the _Mender Python Client_ start looking for updates again.
```

After a succesful update, the _sub-updater_ is responsible for updating the
_artifact_info_ file located in `/etc/mender/artifact_info`, to reflect the name
of the Artifact just installed on the device. This is important, as this is the

> version which is used when polling the server for further updates. The

Reviewer: ```suggestion
version reported to the Mender Server when polling the server for further updates. The
```

`artifact_info` file has to have the structure:

```
artifact_name=<name-of-current-installed-artifact>

> ```

Reviewer: Maybe a link to docs.mender.io?

The `device_type` is taken from `<datadir>/device_type` file, and has to have the structure:

```
device_type=<some-device-type>

> ```

Reviewer: Maybe a link here too?

## Configuration

> The _Client_ respects this subset of configuration variables supported by the original _Mender Client_:

Reviewer: ```suggestion
The _Mender Python Client_ respects this subset of configuration variables supported by the original _Mender Client_:
```

* RootfsPartA
* RootfsPartB
* ServerURL
* ServerCertificate
* TenantToken
* InventoryPollIntervalSeconds
* UpdatePollIntervalSeconds
* RetryPollIntervalSeconds

> 

Reviewer: And where is this configuration expected? In the same /etc/mender/mender.conf? In json format?

I think we should clarify that, and ideally give a link to docs.mender.io for more description of the fields.
